h2/src/main/org/h2/command/Parser.java

```diff
@@ -5687,6 +5686,21 @@ private void initialize(String sql) {
                     checkRunOver(i, len, startLoop);
                 }
                 break;
+            case '?':
+                type = CHAR_SPECIAL_1;
+                if (command[i + 1] == '?') {
+                    char ch = command[i + 2];
```

Comment:
is it possible this can throw IOOBE ? Looks like i+1 is safe, but not sure about i+2

Reply:
The last character in this array is always a space character. Because previous character was a (second) question mark, the next character definitely exists.